client/castApp/src/castinit.c

```diff
@@ -200,6 +200,127 @@ static void addReccasterEnvVarsCallFunc(const iocshArgBuf *args)
+                    num_valid_args--;
+                }
+            }
+        }
+    }
+
+    /* check for duplicates between argv and existing */
+    for (i = 0; i < argc; i++) {
+        for (j = 0; j < self->num_exclude_patterns; j++) {
+            if (argv[i] && strcmp(argv[i], self->exclude_patterns[j]) == 0) {
+                errlogSevPrintf(errlogMinor, "Pattern %s is already in exclude_patterns list\n", argv[i]);
+                argv[i] = NULL;
+                num_valid_args--;
+            }
+        }
+    }
+
+    int num_new_excludes = self->num_exclude_patterns + num_valid_args;
+
+    if (num_valid_args != 0) { // only realloc if new data
+        new_exclude = calloc(num_new_excludes, sizeof(char*)); // alloc bigger
+        if(new_exclude == NULL) {
+            errlogSevPrintf(errlogMajor, "Error in memory allocation of new_exclude from addReccasterExcludePattern\n");
+            epicsMutexUnlock(self->lock);
+            return;
+        }
+
+        /* copy data */
+        for (i = 0; i < self->num_exclude_patterns; i++) {
+            if ((new_exclude[i] = strdup(self->exclude_patterns[i])) == NULL) {
+                errlogSevPrintf(errlogMinor, "strdup error for copying %s to new_exclude[%zu] from addReccasterExcludePattern\n", self->exclude_patterns[i], i);
+                ret = 1;
+                break;
+            }
+        }
+        
+        /* allocate new */
+        if (!ret) {
+            for (count = 0; count < argc; count++) {
+                if (argv[count] != NULL) {
+                    if ((new_exclude[i] = strdup(argv[count])) == NULL) {
+                        errlogSevPrintf(errlogMinor, "strdup error for copying %s to new_exclude[%zu] from addReccasterExcludePattern\n", argv[i], i);
+                        ret = 1;
+                        break;
+                    }
+                    i++;
+                }
+            }
+        }
+        
+        /* pointer swap */
+        char **tmp;
+        tmp = self->exclude_patterns;
+        self->exclude_patterns = new_exclude;
+        new_exclude = tmp;
+
+        /* free */
+        for(i = 0; i < self->num_exclude_patterns; i++) {
+            free(new_exclude[i]);
+        }
+        free(new_exclude);
+        self->num_exclude_patterns = num_new_excludes;
+    }
+    
+    else {
+        errlogSevPrintf(errlogMinor, "No new patterns to add in addReccasterExcludePattern\n");
+    }
+
+    if(ret) {
+        errlogSevPrintf(errlogMajor, "Error in addReccasterExcludePattern - exclude patterns may not be set\n");
+    }
+    epicsMutexUnlock(self->lock);
+}
+
+static const iocshArg addReccasterExcludePatternArg0 = { "excludePattern", iocshArgArgv };
+static const iocshArg * const addReccasterExcludePatternArgs[] = { &addReccasterExcludePatternArg0 };
+static const iocshFuncDef addReccasterExcludePatternFuncDef = {
+    "addReccasterExcludePattern",
+    1,
+    addReccasterExcludePatternArgs
+};
```

Comment:
Another thought: maybe this shouldn't be a varargs-type function, but take a single argument. In my opinion, it is a lot cleaner and easier to follow if someone has
```
addReccasterExcludePattern PAT1
addReccasterExcludePattern PAT2
addReccasterExcludePattern PAT3
```
instead of
```
addReccasterExcludePattern PAT1 PAT2 PAT3
```

Reply:
Sounds good. We will leave with varargs to match addReccasterEnvVars unless someone else has objections